File: ExpandableListSimple/src/main/java/com/expandablelistsimple/expandablelist/SimpleAdapter.java
package com.expandablelistsimple.expandablelist;

import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.expandablelistsimple.R;

import java.util.HashMap;
import java.util.List;

/**
 * by y on 2016/11/21
 */

public class SimpleAdapter extends SimpleBaseExpandableListAdapter<String> {


    public SimpleAdapter(List<String> expandableListTitle, HashMap<String, List<String>> expandableListDetail) {
        super(expandableListTitle, expandableListDetail);
    }

    @Override
    View getGroup(int groupPosition, boolean isExpanded, View convertView, ViewGroup parent) {
        String data = expandableListTitle.get(groupPosition);


        convertView = View.inflate(parent.getContext(), R.layout.group_item, null);

        TextView textView = (TextView) convertView.findViewById(R.id.tv_group);
        ImageView imageView = (ImageView) convertView.findViewById(R.id.iv);

        textView.setText(data);

        if (isExpanded) {
            imageView.setBackgroundResource(R.drawable.dropdown);
        } else {
            imageView.setBackgroundResource(R.drawable.select);
        }

        return convertView;
    }

    @Override
    View getChild(int groupPosition, int childPosition, boolean isLastChild, View convertView, ViewGroup parent) {
        String data = (String) this.getChild(groupPosition, childPosition);

        convertView = View.inflate(parent.getContext(), R.layout.child_item, null);

        TextView tv = (TextView) convertView.findViewById(R.id.tv_child);
        tv.setText(data);
        return convertView;
    }
}

File: utils_expandablelist/src/main/java/sample/util/develop/android/expandablelist/widget/SimpleAdapter.kt
package sample.util.develop.android.expandablelist.widget


import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import sample.util.develop.android.expandablelist.R

class SimpleAdapter(
    expandableListTitle: List<String>,
    expandableListDetail: HashMap<String, List<String>>
) :
    SimpleBaseExpandableListAdapter<String>(expandableListTitle, expandableListDetail) {

    override fun getGroup(
        groupPosition: Int,
        isExpanded: Boolean,
        convertView: View?,
        parent: ViewGroup
    ): View {
        val rootView = View.inflate(parent.context, R.layout.expandable_list_group_item, null)
        val data = expandableListTitle[groupPosition]
        val textView = rootView.findViewById<TextView>(R.id.tvGroup)
        val imageView = rootView.findViewById<ImageView>(R.id.iv)
        textView.text = data
        if (isExpanded) {
            imageView.setImageResource(R.drawable.dropdown)
        } else {
            imageView.setImageResource(R.drawable.select)
        }
        return rootView
    }

    override fun getChild(
        groupPosition: Int,
        childPosition: Int,
        isLastChild: Boolean,
        convertView: View?,
        parent: ViewGroup
    ): View {
        val rootView = View.inflate(parent.context, R.layout.expandable_list_child_item, null)
        val tv = rootView.findViewById<TextView>(R.id.tvChild)
        tv.text = getChild(groupPosition, childPosition)
        return rootView
    }
}
